indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskRunner.java

```diff
@@ -135,4 +135,24 @@ default TaskLocation getTaskLocation(String taskId)
   Map<String, Long> getLazyTaskSlotCount();
 
   Map<String, Long> getBlacklistedTaskSlotCount();
+
+  /**
+   * Beacause the k8s task runner is an extension, we need to know the task runner type in the overlord resource
+   */
+  default boolean isK8sTaskRunner()
```

Comment:
since this class is marked `@PublicApi`, I think we should consider naming this something less specific, like `isDirectTaskRunner` or `isWorkerFreeTaskRunner` something to indicate that it is worker free but can still run up to `getTotalTaskSlotCount` tasks (based on how its being used in `OverlordResource`)